MechJeb2/MechJebModuleStagingController.cs

```diff
@@ -94,23 +94,27 @@ public override void OnUpdate()
             if (HasStayingChutes(StageManager.CurrentStage - 1, vessel))
                 return;
 
-            //only fire decouplers to drop deactivated engines or tanks or fairings
-            bool firesDecoupler = InverseStageFiresDecoupler(StageManager.CurrentStage - 1, vessel);
-            if (firesDecoupler && !(InverseStageDecouplesDeactivatedEngineOrTank(StageManager.CurrentStage - 1, vessel) || HasFairing(StageManager.CurrentStage - 1, vessel)))
-                return;
-
-            //only decouple fairings if the dynamic pressure and altitude conditions are respected
-            if ((core.vesselState.dynamicPressure > fairingMaxDynamicPressure || core.vesselState.altitudeASL < fairingMinAltitude) &&
-                HasFairing(StageManager.CurrentStage - 1, vessel))
-                return;
+            //Always drop deactivated engines or tanks
+            if (!InverseStageDecouplesDeactivatedEngineOrTank(StageManager.CurrentStage - 1, vessel))
+            {
+                //only decouple fairings if the dynamic pressure and altitude conditions are respected
+                if ((core.vesselState.dynamicPressure > fairingMaxDynamicPressure || core.vesselState.altitudeASL < fairingMinAltitude) &&
+                    HasFairing(StageManager.CurrentStage - 1, vessel))
+                    return;
+
+                //only release launch clamps if we're at nearly full thrust
+                if (vesselState.thrustCurrent / vesselState.thrustAvailable < 1/1.2 &&

```

Comment:
I wouldn't think too hard right this second about Engine failures, I was sort of throwing that out there as a thing to consider how it would get dealt with eventually.  Probably something needs to get added to VesselState to determine if all the active engines are healthy or not (the icon turns yellow/red, so presumably there's some state there MechJeb could peek at directly), then it would reduce to just checking a boolean condition here.

I think the road you're heading down is roughly what I thought about and found no clear obviously correct answers, which was leading me towards wanting it to be tweakable in the UI -- and I would agree UI sprawl should be avoided if possible, but I can't quite see the obviously correct answer.